llmfoundry/data/finetuning/dataloader.py

```diff
@@ -152,6 +152,7 @@ def build_finetuning_dataloader(cfg: DictConfig,
             sampling_method=cfg.dataset.get('sampling_method', 'balanced'),
             sampling_granularity=cfg.dataset.get('sampling_granularity', 1),
             batching_method=cfg.dataset.get('batching_method', 'random'),
+            max_seq_len=cfg.dataset.max_seq_len,
```

Comment:
Is `cfg.dataset.max_seq_len` a mandatory argument? What happens if `max_seq_len` is missing from `cfg.dataset`?

Reply:
if it's missing, it will be using the default 2048. I added this because it's needed when convert to list from buffer in  `np.from_buffer(sample['input_ids'], dtype=np.int64)[:self.max_seq_len].tolist()`. I guess the `[:self.max_seq_len]` is not needed if the data already truncates at `max_seq_len` when generating the MDS file ?

Reply:
I think we should do it the way you implemented it. This matches the behavior of `text_data.py`, and allows you to run with data truncated to the requested max seq len.